test/e2e/user_group_limit/user_group_limit_test.go

```diff
@@ -911,6 +925,206 @@ var _ = ginkgo.Describe("UserGroupLimit", func() {
+							Limit:           "user entry",
+							Users:           []string{user1},
+							MaxApplications: 1,
+							MaxResources: map[string]string{
+								siCommon.Memory: fmt.Sprintf("%dM", mediumMem),
+							},
+						},
+						{
+							Limit:           "user2 entry",
+							Users:           []string{user2},
+							MaxApplications: 2,
+							MaxResources: map[string]string{
+								siCommon.Memory: fmt.Sprintf("%dM", largeMem),
+							},
+						},
+					}})
+				if err != nil {
+					return err
+				}
+				return common.AddQueue(sc, constants.DefaultPartition, constants.RootQueue, configs.QueueConfig{Name: "sandbox2"})
+			})
+		})
+		// Backup the existing kubeconfig
+		oldKubeconfigPath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+		if _, err := os.Stat(oldKubeconfigPath); !os.IsNotExist(err) {
+			oldKubeconfigContent, err = os.ReadFile(oldKubeconfigPath)
+			gomega.Ω(err).NotTo(HaveOccurred())
+		}
+		// Create Service Account
+		ginkgo.By("Creating Service Account...")
+		_, err := clientset.CoreV1().ServiceAccounts(namespace).Create(context.TODO(), &v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: serviceAccountName,
+			},
+		}, metav1.CreateOptions{})
+		gomega.Ω(err).NotTo(HaveOccurred())
+		// Create a ClusterRole with necessary permissions
+		ginkgo.By("Creating ClusterRole...")
+		clusterRole := &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod-creator-role",
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods", "serviceaccounts"},
+					Verbs:     []string{"create", "get", "list", "watch", "delete"},
+				},
+			},
+		}
+		_, err = kClient.CreateClusterRole(clusterRole)
+		gomega.Ω(err).NotTo(HaveOccurred())
+		// Create a ClusterRoleBinding to bind the ClusterRole to the service account
+		ginkgo.By("Creating ClusterRoleBinding...")
+		clusterRoleBinding := &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod-creator-role-binding",
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     "pod-creator-role",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      "test-user-sa",
+					Namespace: "default",
+				},
+			},
+		}
+		_, err = kClient.CreateClusterRoleBinding(clusterRoleBinding.ObjectMeta.Name, clusterRoleBinding.RoleRef.Name, clusterRoleBinding.Subjects[0].Namespace, clusterRoleBinding.Subjects[0].Name)
+		gomega.Ω(err).NotTo(HaveOccurred())
+		// Create a Secret for the Service Account
+		ginkgo.By("Creating Secret for the Service Account...")
+		_, err = clientset.CoreV1().Secrets(namespace).Create(context.TODO(), &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secretName,
+				Annotations: map[string]string{
+					"kubernetes.io/service-account.name": serviceAccountName,
+				},
+			},
+			Type: v1.SecretTypeServiceAccountToken,
+		}, metav1.CreateOptions{})
```

Comment:
Please use `KubeCtl.CreateSecret()` if possible. If the annotation is necessary, create a new method with the name `CreateSecretWithAnnotation(secret *v1.Secret, namespace string, annotations map[string]string)` to avoid calling this directly.

Reply:
Updated the code with requested changes.